backend/libinput/pointer.c

```diff
@@ -16,7 +16,8 @@ struct wlr_pointer *create_libinput_pointer(
 		wlr_log(WLR_ERROR, "Unable to allocate wlr_pointer");
 		return NULL;
 	}
-	wlr_pointer_init(wlr_pointer, NULL);
+	wlr_pointer_init(wlr_pointer, NULL,
+		libinput_device_config_tap_get_finger_count(libinput_dev) > 0);
```

Comment:
I think I would rather fish this interface through than use it as the basis for heuristics.

Reply:
What about touchpads which support just one finger?

Reply:
> I think I would rather fish this interface through than use it as the basis for heuristics.

What do you mean by fishing it through the interface? Add another argument to create_libinput_pointer?

> What about touchpads which support just one finger?

I used what [clutter uses](https://gitlab.gnome.org/GNOME/mutter/blob/master/clutter/clutter/evdev/clutter-input-device-evdev.c#L1459). I'm not too familiar with the workings of libinput, but if a touchpad only supports one finger, shouldn't `libinput_device_config_tap_get_finger_count` return 1?

Reply:
No, I mean add a wlr_pointer_get_finger_count or something similar.

I think I'd sooner just not support this feature. I don't like heursitic driven features like this.

Reply:
Fair enough.

This was something that I wanted so that I could configure sway input devices by device type. I think it would be helpful to a lot of people who put their configs in a dotfile repo and would like to have one config work on multiple devices.

I also did some more research on what libinput did and it doesn't seem like there's any other way to determine if an input device is a touchpad currently.

Reply:
Heuristics should live in the compositor. We do expose libinput objects.